Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -682,6 +683,35 @@ void WebResourceLoadStatisticsStore::setAppBoundDomains(HashSet<RegistrableDomai
 }
 #endif
 
+#if ENABLE(MANAGED_DOMAINS)
+void WebResourceLoadStatisticsStore::setManagedDomains(HashSet<RegistrableDomain>&& domains, CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+
+    if (isEphemeral() || domains.isEmpty()) {
```

Comment:
Why is it OK to silently do nothing if someone passes in an empty set? What if the set in the statistics store is not already empty?

Reply:
We don't want to change the cookie blocking mode if it's an empty set. The set being empty implies there's no managed websites.

Reply:
Sure, but what if someone first set it to a non-empty set and is now setting it back to the empty set. Does the code handle that case properly? Or is that a forbidden sequence of calls?